### ISSUE TITLE
Docker image auto build and deploy

### ISSUE THREAD

**Issue by kfeofantov** [gohugoio/hugo#4693]
Please, add automated docker build and push at docker hub in CI Pipeline. 
I can write pipeline for CircleCI or Travis.

**Comment by jhabdas** [comment]
I was looking on Kitematic yesterday and there're quite a few _unofficial_ dockers for hugo, most of which are [not well-maintained](https://github.com/ecma6/hugo) or poorly constructed (i.e. not using scratch).

The current `Dockerfile` in the project repo makes some assumptions, such as to which ports to expose and volumes to use, which, while nice, would still require some adjustments for use in some [Compose](https://docs.docker.com/compose/) projects.

Given the base image, env vars and apt packages will change between builds having an official image will help prevent the need to use something like Docker-in-Docker, which adds a funky abstraction to running hugo in containerized environments.

**TL;DR:** Please update the CI/CD for deploys to DockerHub as OP suggests.

**Comment by jhabdas** [comment]
For those looking for a quick way to build the `0.44` with Docker I've created a Dockerfile to build tagged versions of Hugo here (path subject to change):

https://git.habd.as/comfusion/after-dark/src/branch/fathom/docker/hugo/Dockerfile

**Comment by mlake** [comment]
+1 for Composeability

**Comment by Yajo** [comment]
I think this is still relevant because Hugo users usually perform some kinda autobuild for their websites, usually done in a CI/CD pipeline, usually based on Docker for easiness and reusability.

This would make the project more usable, and still more maintainable because CI would do the dirty job.

**Comment by jonahbron** [comment]
I would also like to see an officially maintained Docker image be added. :+1:

**Comment by eugene-bright** [comment]
There is [an official docker image on dockerhub][hub]. But it is stale like this issue.

[hub]: https://hub.docker.com/r/gohugoio/hugo/tags

**Comment by bep** [comment]
>There is an official docker image on dockerhub. But it is stale like this issue.

That `gohugoio` account isn't official.

**Comment by eugene-bright** [comment]
So it's a well crafted fake that drains reputation then.

**Comment by pierreozoux** [comment]
Gitlab provides one.
I also think that the project should provide one, it would be cleaner for our CICD pipelines :)

**Comment by jufemaiz** [comment]
LGTM on dockerhub?

**Comment by ChenXiyu** [comment]
It is a good practice to wrap a command line tool into a docker image, I believe. 
Could you please release an official image for us?
Or you have other concerns about this?

**Comment by peaceiris** [comment]
> We (I) plan to use GitHub to publish docker images.
>
> https://github.com/gohugoio/hugo/pull/6405#issuecomment-548280568

FYI.

**Comment by Christoph-Harms** [comment]
How can one help with this? Sure, if people want to use hugo in their automated, containerized environments, they can just clone the repo and do the docker build itself. But it is pretty much standard to provide a dockerized version of an application. It's the direction the industry has been moving in for the past years.

When I tried to use the `gohugoio/hugo` image last week and noticed that it is based upon a version of hugo that is two years old, I was very confused. Then I noticed that it is not THIS repo that the image is built upon, but a fork. So I thought: "Alright, some 2 years ago, someone broke the docker build. And apparently it was very hard to fix or something. So they switched to a working fork, and then forgot it or deprioritized it or whatever." So I thought, I'm going to fork that repo and fix the damn build, do something for the OS community.

I was even more confused when I cloned my fork, ran `docker build`  on it and everything worked. I don't understand. Docker Hub as an autobuild feature. Providing an up-to-date docker image should be as easy as re-connecting THIS repo to the `gohugoio/hugo` repo  on Docker Hub and telling it to auto-build everytime something gets merged into `master`. But then  again, if it was  that easy, it would have been done, right? So what am I missing?

**Comment by bep** [comment]
@Christoph-Harms I recommend this one: https://hub.docker.com/r/klakegg/hugo/tags

**Comment by Christoph-Harms** [comment]
> That `gohugoio` account isn't official.

I overlooked that comment of yours, @bep. That explains it. Shame that it "looks" official, because the account names on Docker Hub and Github are the same. :/ It gives a taste of, I don't know, unreliability? It's not exactly what I mean, but I don't have a better word. It puts hugo in a bad light. Kind of. I hope what I want to say here actually comes across. 😅 I love hugo and I want everyone else to love it, too! :)

> @Christoph-Harms I recommend this one: https://hub.docker.com/r/klakegg/hugo/

Looks great, thank you! I didn't even find that one when browsing Docker Hub. Only old forks or specialized builds.

**Comment by bep** [comment]
@Christoph-Harms @klakegg is reliable. I understand that the "official Hugo docker situation" isn't ... ideal, but it's a priority/time thing. I have now added the klakegg Docker to the Hugo install page so it should be easier to find.
